src/items/external-blocks.md

```diff
@@ -29,9 +29,13 @@
 > _NamedFunctionParametersWithVariadics_ :\
 > &nbsp;&nbsp; ( _NamedFunctionParam_ `,` )<sup>\*</sup> _NamedFunctionParam_ `,` `...`
 
-External blocks form the basis for Rust's foreign function interface.
-Declarations in an external block describe symbols in external, non-Rust
-libraries.
+External blocks provide _declarations_ of items that are not _defined_ in the
+current crate and are the basis of Rust's foreign function interface. These are
+sort of like unchecked imports. 
```

Comment:
Nit: "sort of like" reads rather like spoken English (not something you'd find in an IEEE paper...).

Reply:
I didn't intend to take @eddyb feedback literally here, but that's the best I could come up with. 

These blocks declare that something exists in a particular way. If you "use" (call, access, etc.) it and it doesn't exist, or it exists in a way that's different than how you declared it: boom. If you don't actually use it, whether it exists doesn't matter, and if it doesn't, that's ok.

OTOH, to me, an import is an operation that brings something into scope that exists. If it doesn't exist, then you can't import it. So I'm not sure what would be the best way to make an analogy with imports here.

Maybe something like

> Declarations are similar to imports, in that they bring an item into scope. The main difference is that an import requires an item to exist, while a declaration states that an item exists with a particular type signature, path, etc. As long as a declared function is not called or a declared static is not accessed, these declared items do not need to exist. This allows, for example, declaring a dynamically-linked function unconditionally, as long as it is only used when the function is actually dynamically linked. 

?

Reply:
Maybe with an example like this:

```rust
extern "C" { fn foo(); }

fn main() {
    let x = foo;
    if symbol_of_foo_not_linked() {
         link_symbol_of_foo();
    }
    unsafe { x() };
}
```

?

Reply:
@eddyb @ehuss wdyt about the wording ^-- ?

Reply:
I use "FFI import" to mean something different from what the `use` item does in Rust, so I'm not comparing the two but rather using the more general meaning of "import".

Also, not sure that where the function comes from in terms of linkage is *that* important, compared to the fact that *it likely isn't even written in Rust*, hence "FFI".